fecgov/openFEC
# Create endpoint to calculate total electioneering communications

## Summary (required)

- Resolves #[_4062_](https://github.com/fecgov/openfec/issues/4062)

_New endpoint is created to calculate total electioneering communications. Three filters are added to the endpoint. None of them is required_
  - election_full:  true by default
  - candidate_id
  - cycle  ( mapping to different columns depending on the value of election_full )

## How to test the changes locally

- Download feature branch
- Run pytest to make sure all tests pass
- Run api on local with setting SQLA_CONN to DEV database
- Returning total electioneering communications on each candidate per cand_eelction_year
http://127.0.0.1:5000/v1/electioneering/totals/by_candidate/?page=1&sort_null_only=false&per_page=20&sort_nulls_last=false&election_full=true&sort_hide_null=false

- Returning total electioneering communications on each candidate per fec_cycle
http://127.0.0.1:5000/v1/electioneering/totals/by_candidate/?page=1&sort_null_only=false&per_page=20&sort_nulls_last=false&election_full=false&sort_hide_null=false

- Returning total electioneering communications on P80001571 per fec_cyce
http://127.0.0.1:5000/v1/electioneering/totals/by_candidate/?page=1&sort_null_only=false&per_page=20&sort_nulls_last=false&election_full=false&sort_hide_null=false&candidate_id=P80001571

- Returning total electioneering communications on H0NC09187 on 2020
http://127.0.0.1:5000/v1/electioneering/totals/by_candidate/?cycle=2020&page=1&sort_null_only=false&per_page=20&sort_nulls_last=false&election_full=false&sort_hide_null=false&candidate_id=H0NC09187
